cmd/thanos/compact.go

```diff
@@ -499,6 +500,9 @@ func (cc *compactConfig) registerFlag(cmd *kingpin.CmdClause) *compactConfig {
 		"This works well for deduplication of blocks with **precisely the same samples** like produced by Receiver replication.").
 		Hidden().StringsVar(&cc.dedupReplicaLabels)
 
+	cmd.Flag("download.retries", "How many time to retry downloading a block with exponential backoff before giving up").
```

Comment:
hm again, this is similar to the https://github.com/thanos-io/thanos/pull/2908 discussion. Shouldn't this be applied by upload/download client itself?